### DISCUSSION TITLE
[10. 13] 리포트, 역량 파트 회의

### DISCUSSION THREAD

**Discussion by sunhpark42** [woowacourse/prolog#486] [Feature]
# 2021. 10. 13 회의록
1. 역량
	1. 상위역량 삭제 시 하위역량 전체 삭제
		1. 현재 : 상위역량이 하위 역량을 1개 이상 가질 때 삭제 할 수 없음 (ERROR)
	2. 등록된 역량이 없을 시
		1. 프론트 / 백엔드 선택하면 각 영역의 디폴트 역량이 자동추가 되는 기능추가
			1. 백엔드 API 필요.
	3. 역량 목록
	4. URL path
		1. /ability ? /abilities ?
		2. 일단 현상 유지  /ability

2. 리포트
	1. URL path
		1. Report/write -> reports/write로 변경
	2. 리포트 생성 시 학습로그가 없을 때
		* Confirm 넣기
	3. 리포트 페이지 메인
		1. 리포트 목록페이지 추가.
		2. 개별 리포트 페이지는 유지
	4. 리포트 역량 그래프 추가. 
	5. 대표 리포트
		1. 레벨별 리포트 4개
		2. + 전체 대상 리포트
			1. Ex) 내가 나를 표현하는 주제로(접근성, CS 등) 리포트 작성해서 대표 등록.
		3. 나중에 pdf뽑기, 공유하기 기능 추가 되면 좋겠다.
	6. 리포트 작성 및 수정 시 학습로그 추가에 레벨 선택항목에 `전체레벨`  , `모두 선택` 추가
	7. 리포트 API 분리 -> later
		1. PUT /reports
		2. GET /reports를
			1. GET /reports/{:id}/{:No-name}
				* 제목, 날짜, 설명 등
			2. GET /reports/{:id}/abilityGraph
				* 역량 그래프
			3. GET /reports/{:id}/studylog 
				* 학습로그 목록
				* 페이지네이션 추가
			* 으로 분리
	8. 리포트 학습로그 목록의 역량 칸
		1. 호버시에 스크롤 바 보이게 수정
		2. 각 학습로그 역량 같은 범주 이내로 정렬.
		

3. 개인 페이지 > 학습로그 목록
	1. 한 페이지 5개씩

4. 오버뷰
	1. GitHub README 같은 부분이 있으면 좋겠다.
		1. HOW?
			* 대표 리포트의 그래프 + description의 축소판을 넣기.

6. 메인페이지
	1. 학습로그 노출 시 제목 + 컨텐츠 모두 노출.

* 좋은 익스텐션
	* VisuBug : 크롬 익스텐션, 바로 css 속성 확인 가능함

**Comment by devhyun637** [comment]
감사합니다! 회의정리!

**Comment by gracefulBrown** [comment]
정리 고마워요~

**Comment by gracefulBrown** [comment]
- 리포트 개선
  - https://github.com/woowacourse/prolog/issues/489
- 리포트 역량 관리 (일부 수정)
  - https://github.com/woowacourse/prolog/issues/247
- 역량 개선
  - https://github.com/woowacourse/prolog/issues/488
- 프로필 > 학습로그
  - https://github.com/woowacourse/prolog/issues/492
- 오버뷰 대표 리포트
  - https://github.com/woowacourse/prolog/issues/491
- 메인 페이지
  - https://github.com/woowacourse/prolog/issues/490
